Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -4,10 +4,20 @@ import SwiftBSON
 /// Represents a MongoDB connection string.
 /// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/
 public struct MongoConnectionString: Codable, LosslessStringConvertible {
+    /// Characters that must not be present in a database name.
     private static let forbiddenDBCharacters = ["/", "\\", " ", "\"", "$"]
-    /// Forbidden characters per RFC 3986.
+
+    /// General delimeters as defined by RFC 3986. These characters must be percent-encoded when present in the hosts,
+    /// default authentication database, and user info.
     /// - SeeAlso: https://datatracker.ietf.org/doc/html/rfc3986#section-2.2
-    fileprivate static let forbiddenUserInfoCharacters = [":", "/", "?", "#", "[", "]", "@"]
+    fileprivate static let genDelims = ":/?#[@"
```

Comment:
are we missing `]` here now? (it's on the RFC list)

Reply:
oops good catch, fixed